packages/repository-tests/src/crud/relations/acceptance/has-many-inclusion-resolver.relation.acceptance.ts

```diff
@@ -0,0 +1,273 @@
+
+import {expect, skipIf, toJSON} from '@loopback/testlab';
+import {Suite} from 'mocha';
+import {
+  CrudFeatures,
+  CrudRepositoryCtor,
+  CrudTestContext,
+  DataSourceOptions,
+} from '../../..';
+import {
+  deleteAllModelsInDefaultDataSource,
+  MixedIdType,
+  withCrudCtx,
+} from '../../../helpers.repository-tests';
+import {
+  Customer,
+  CustomerRepository,
+  Order,
+  OrderRepository,
+} from '../fixtures/models';
+import {givenBoundCrudRepositories} from '../helpers';
+
+export function hasManyRelationAcceptance(
+  dataSourceOptions: DataSourceOptions,
+  repositoryClass: CrudRepositoryCtor,
+  features: CrudFeatures,
+) {
+  skipIf<[(this: Suite) => void], void>(
+    !features.supportsInclusionResolvers,
+    describe,
+    'retrieve models including relations',
+    () => {
+      describe('HasMany inclusion resolvers - acceptance', () => {
+        before(deleteAllModelsInDefaultDataSource);
+        let customerRepo: CustomerRepository;
+        let orderRepo: OrderRepository;
+        let existingCustomerId: MixedIdType;
+
+        before(
+          withCrudCtx(async function setupRepository(ctx: CrudTestContext) {
+            // when running the test suite on MongoDB, we don't really need to setup
+            // this config for mongo connector to pass the test.
+            // however real-world applications might have such config for MongoDB
+            // setting it up to check if it works fine as well
+            Order.definition.properties.customerId.type = features.idType;
+            Order.definition.properties.customerId.mongodb = {
+              dataType: 'ObjectID',
+            };
+            // this helper should create the inclusion resolvers for us
+            ({customerRepo, orderRepo} = givenBoundCrudRepositories(
+              ctx.dataSource,
+              repositoryClass,
+            ));
+            // inclusionResolvers should be defined. And resolver for each
+            // relation should be created by the hasManyFactory at this point.
+            expect(customerRepo.inclusionResolvers).to.not.be.undefined();
+            expect(orderRepo.inclusionResolvers).to.not.be.undefined();
+            expect(customerRepo.orders.inclusionResolver).to.not.be.undefined();
+            expect(
+              customerRepo.customers.inclusionResolver,
+            ).to.not.be.undefined();
+            // inclusionResolvers shouldn't setup yet at this point
+            expect(customerRepo.inclusionResolvers).to.deepEqual(new Map());
+
+            await ctx.dataSource.automigrate([Customer.name, Order.name]);
+          }),
+        );
+
+        beforeEach(async () => {
+          customerRepo.inclusionResolvers.set(
+            'orders',
+            customerRepo.orders.inclusionResolver,
+          );
+          customerRepo.inclusionResolvers.set(
+            'customers',
+            customerRepo.customers.inclusionResolver,
+          );
```

Comment:
IMO, setup of inclusion resolvers belongs to `givenBoundCrudRepositories`.

Conceptually, in a real application, when we have a model-specific repository class like `CustomerRepository`, then inclusion resolvers are registered from the constructor. In our tests here, we use factory functions like `givenBoundCrudRepositories` as a replacement for model-specific classes, therefore initialization of inclusion resolvers belongs there too.